gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementStateStore.java

```diff
@@ -36,6 +40,18 @@
  */
 @Alpha
 public interface DagManagementStateStore {
+  /**
+   * Returns a {@link FlowSpec} for the given URI.
+   * @throws SpecNotFoundException if the spec is not found
+   */
+  FlowSpec getFlowSpec(URI uri) throws SpecNotFoundException;
+
+  /**
+   * Removes a {@link FlowSpec} with the given URI and pass the deletion to listeners if `triggerListener` is true
+   * No-op if the flow spec was not present in the store.
+   */
+  void remove(URI uri, Properties headers, boolean triggerListener);
```

Comment:
I like the name `getFlowSpec`. should this be `removeFlowSpec` to parallel that?